Fix field escape parsing error in csvparser

rbuckton opened the pull request:
This fixes a bug in `CsvParser.prototype.escapeField` that improperly handles single-character escapes like `\\`, which affects parsing paths on DOS-based file systems.

rbuckton committed:
Fix field escape parsing error in csvparser

thlorenz commented:
Thanks, but as indicated in the readme these tools are extracted from the [v8 repo](https://github.com/v8/v8).
I'd like to keep feature (and also bug) parity with that repo, so could you please file a PR there and then we can upstream the fix here?

Additionally please remove the `package.lock.json` from the PR as it has nothing to do with the fix. For libraries I prefer not to lock dependencies to get bug fixes applied automatically.

Thanks.

rbuckton commented:
I've filed https://github.com/v8/v8/pull/36 against v8. I've also removed package-lock.json and added an .npmrc that will prevent generation of package-lock.json files in the future.

rbuckton committed:
Update to latest version from v8/tools

rbuckton committed:
Update csvparser.js

rbuckton commented:
I've reverted my change to this file and instead updated it to the latest version from https://github.com/v8/v8/blob/master/tools/csvparser.js which already has a compatible fix.

rbuckton committed:
Fix 'ConsArray' reference
